### PR TITLE
feat(rules): Minor formatting and copy changes

### PR DESCRIPTION
**Pull request by ceorourke**
Final touches before EAing the Slack notification confirming issue alert rule creation and edits - change bold items to be in single quotes and change the create confirmation notification text to be "...and will send notifications here" to "...and will send notifications to this channel". 

Screenshot 2024-06-13 at 2 45 35 PM

### CODE REVIEW

File: src/sentry/rules/actions/utils.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def get_changed_data(
             pass
 
         if environment:
-            changed_data["environment"].append(f"Removed *{environment.name}* environment")
+            changed_data["environment"].append(f"Removed '{environment.name}' environment")
```
Comment (saponifi3d) [inline src/sentry/rules/actions/utils.py:134]: The logic here and in the previous if statement feel incredibly similar.. could this be simplified a bit and then we use a helper function to decide if there's an update required?

for example:
```python
added_env = get_environment_change(rule_data, rule_data_before)
before_env = get_environment_change(rule_data_before, rule_data)

if added_env:
  ...
  
if before_env:
  ...
```
This allows us to simplify the code / remove the duplicate logic. it also would match the way the other if statements work.. (could likely put those into a dictionary and enumerate each change data in a for loop then, drastically simplify this code)

```python
data_to_check = {
   label_text: check_value_changed(rule_data, rule_data_before, "label", "rule name"),
   action_match: check_value_changed(rule_data, rule_data_before, "action_match", "trigger"),
   ...
}

for key, value in data_to_check:
  if value:
     changed_data[key] = value
```

**Review by saponifi3d — APPROVED** [review]
left a few comments on how we could tidy this code up a little more for our future selves, but take them as a nit :)

File: src/sentry/integrations/slack/message_builder/notifications/rule_save_edit.py [hunk 2]
```diff
@@ -39,12 +42,14 @@ def get_settings_url(self):
...
+        org_slug = self.rule.project.organization.slug
+        project_slug = self.rule.project.slug
+        rule_url = self.linkify_rule(org_slug, project_slug)
+        project_url = self.linkify_project(org_slug, project_slug)
         if self.new:
-            rule_text = f"Alert rule {rule_url} was created in the *{project}* project and will send notifications here."
+            rule_text = f"Alert rule {rule_url} was created in the {project_url} project and will send notifications to this channel."
         else:
-            rule_text = f"Alert rule {rule_url} in the *{project}* project was updated."
+            rule_text = f"Alert rule {rule_url} in the {project_url} project was updated."
```
[the remark this answers is not in the document]
Comment (ceorourke) [reply src/sentry/integrations/slack/message_builder/notifications/rule_save_edit.py:52]: ?

[the remark this answers is not in the document]
Comment (saponifi3d) [reply src/sentry/integrations/slack/message_builder/notifications/rule_save_edit.py:48]: internationalization - if we're showing text to a user and they have their language configured to german, the UI could all be in german then this message could show up in english. not sure if that's a concern or not.

**Pull request merged**